### PR TITLE
Cor 125 iam api validation

### PR DESCRIPTION
**Pull request by nilueps**

### CODE REVIEW

File: api/pkg/apps/iam/attribute_post.go [hunk 1]
```diff
@@ -18,6 +19,18 @@ func (s *Server) postAttributes(w http.ResponseWriter, req *http.Request) {
...
 	}
 
+	errList := ValidateAttribute(&a, validation.NewPath(""))
+	if len(errList) > 0 {
+		status := validation.Status{
+			Status:  validation.Failure,
+			Code:    http.StatusUnprocessableEntity,
+			Message: "invalid attribute",
+			Errors:  errList,
+		}
```
Comment (ludydoo) [inline api/pkg/apps/iam/attribute_post.go:29]: Would probably make sense to extract this
```
func NewValidationErr(message string, errs ErrList)
```

File: api/pkg/apps/iam/attribute_validation.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+				err := validation.Required(translationPath.Child("long"), "Long formulation is required")
+				errs = append(errs, err)
+			}
+			// Validate ShortFormulation
+			if len(translation.ShortFormulation) == 0 {
+				err := validation.Required(translationPath.Child("short"), "Short formulation is required")
+				errs = append(errs, err)
+			}
+		}
+	}
```
Comment (ludydoo) [inline api/pkg/apps/iam/attribute_validation.go:50]: This should be its own method..

```
func ValidateTranslation(translation *Translation, path validation.Path, errList validation.ErrList) ErrList
```

File: api/pkg/apps/iam/individual_validation.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package iam
+
+import "github.com/nrc-no/core/pkg/validation"
+
+func ValidateIndividual(individual *Individual, path *validation.Path) validation.ErrorList {
+	// Validate Party
+	errs := ValidateParty(individual.Party, path)
+
```
Comment (ludydoo) [inline api/pkg/apps/iam/individual_validation.go:8]: We should also make sure that the party has the `Individual` party type

File: api/pkg/apps/iam/party_validation.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	}
+
+	// Validate Attributes
+	//	// For now Party Attributes are just a map[string][]string but eventually
+	//	// we will want TODO some changes to have more information about value types
+	//	// and validation
+	//attributesPath := path.Child("attributes")
+	//for name, values := range party.Attributes {
+	//	attributePath := attributesPath.Key(name)
+	//}
```
Comment (ludydoo) [inline api/pkg/apps/iam/party_validation.go:21]: Can you remove the comment and create a jira issue ?

File: api/pkg/validation/errors.go [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+// Error implements the error interface.
+func (v *Error) Error() string {
+	return fmt.Sprintf("%s: %s", v.Field, v.ErrorBody())
+}
+
+// ErrorBody returns the error message without the field name.  This is useful
+// for building nice-looking higher-level error reporting.
+func (v *Error) ErrorBody() string {
+	var s string
```
Comment (ludydoo) [inline api/pkg/validation/errors.go:48]: I think you can just do `fmt.Sprintf("%#v`, v)`

**Review by ludydoo — CHANGES_REQUESTED** [review]

File: api/pkg/validation/errors.go (hunk 5 above)
Comment (nilueps) [reply api/pkg/validation/errors.go:48]: Not sure what you mean here. Should I overwrite the entire method with that oneliner?
It's not my code btw, its from kubernetes

File: api/pkg/apps/webapp/casetypes.go [hunk 6]
```diff
@@ -17,26 +17,24 @@ func (s *Server) CaseTypes(w http.ResponseWriter, req *http.Request) {
...
 
-	if req.Method == "POST" {
-		s.PostCaseType(ctx, &cms.CaseType{}, w, req)
-		return
-	}
-
 	caseTypes, err := cmsClient.CaseTypes().List(ctx, cms.CaseTypeListOptions{})
 	if err != nil {
-		s.Error(w, err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (ludydoo) [inline api/pkg/apps/webapp/casetypes.go:22]: dont change that booby

File: api/pkg/apps/iam/attribute_post.go [hunk 7]
```diff
@@ -18,6 +19,13 @@ func (s *Server) postAttributes(w http.ResponseWriter, req *http.Request) {
 		a.ID = uuid.NewV4().String()
 	}
 
+	errList := ValidateAttribute(&a, validation.NewPath(""))
+	if len(errList) > 0 {
+		status := errList.Status(http.StatusUnprocessableEntity, "invalid attribute")
+		s.json(w, status.Code, status)
```
Comment (ludydoo) [inline api/pkg/apps/iam/attribute_post.go:25]: s.Error(status)

**Review by ludydoo — APPROVED** [review]
Plz just change the http.Error(...)  to s.Error(w, err)

**Pull request merged**